### DISCUSSION TITLE
Configuration Versioning

### DISCUSSION THREAD

**Discussion by jdhitsolutions** [mdgrs-mei/PowerShellRun#68] [Ideas]
Since a user needs to write a configuration script to use this module, I wonder if there is any benefit in providing a versioning mechanism. As I'm developing and revising my configuration, it would be helpful to know what version I'm currently using. I'm thinking of a hot key I could press to show version info. Maybe the user could define variables in their script like `$PSRunVersion` and `$PSRunConfiguration` to show the configuration file.

It would be nice to get some feedback on this idea from other module users.

**Comment by jdhitsolutions** [comment]
For now, I'm defining variables in my configuration script:

```powershell
Set-Variable -Name PSRunVersion -Value '2.1.0' -Scope Global
Set-Variable -Name PSRunAuthor -Value 'Jeff Hicks' -Scope Global
Set-Variable -Name PSRunConfiguration -Value $MyInvocation.MyCommand.Path -Scope Global
Set-Variable -Name PSRunConfigurationDate -Value (Get-Item $MyInvocation.MyCommand.Path).LastWriteTime -Scope Global
```

And then defining a function.

```powershell
Start-PSRunFunctionRegistration

Function global:Get-PSRunVersion {
    [PSCustomObject]@{
        Version       = $PSRunVersion
        Author        = $PSRunAuthor
        Configuration = $PSRunConfiguration
        LastModified  = $PSRunConfigurationDate
    } | Out-Host
   #reload
    psrun
}
```

![image](https://github.com/user-attachments/assets/d493d532-8389-4d1f-989d-1bdb6794707a)

`PSRun` is an alias for `Invoke-PSRun` defined in my configuration.

This will only be useful once we find a way for defined functions to always be displayed.

**Reply by mdgrs-mei** [reply]
You can also use custom `Description` or `Preview` string to show the version.

```powershell
$PSRunVersionLine = "$PSRunVersion, $PSRunAuthor, $PSRunConfigurationDate"
Add-PSRunScriptBlock -Name 'Version' -Description $PSRunVersionLine -ScriptBlock {
}
```

![image](https://github.com/user-attachments/assets/81428ed5-7dc7-4f83-aad9-56c69653d152)

**Reply by jdhitsolutions** [reply]
That is a handy tip.
